packages/template-retail-react-app/app/components/search/index.jsx

```diff
@@ -151,6 +165,47 @@ const Search = (props) => {
         setIsOpen(false)
     }
 
+    useEffect(() => {
+        const handleEmbeddedMessageSent = (e) => {
+            if (!miawChatRef.current.hasFired && miawChatRef.current.newChatLaunched) {
+                if (
+                    e.detail.conversationEntry?.sender?.role === 'Chatbot' &&
+                    searchInputRef?.current?.value
+                ) {
+                    miawChatRef.current.hasFired = true
+                    setTimeout(() => {
+                        window.embeddedservice_bootstrap.utilAPI.sendTextMessage(
+                            searchInputRef.current.value.trim()
+                        )
+                    }, 500)
+                }
+            }
+        }
+
+        window.addEventListener('onEmbeddedMessageSent', handleEmbeddedMessageSent)
+
+        return () => {
+            // Clean up
+            window.removeEventListener('onEmbeddedMessageSent', handleEmbeddedMessageSent)
+        }
+    }, [])
+
+    const launchChat = () => {
+        window.embeddedservice_bootstrap.utilAPI
+            .launchChat()
+            .then((successMessage) => {
+                /* TODO: With the Salesforce Winter '26 release, we will be able to use the
+                 * onEmbeddedMessagingFirstBotMessageSent event instead, and get rid of this logic. */
+                if (successMessage.includes('Successfully initialized the messaging client')) {
```

Comment:
I'm sure you've tried to find another way around this, but WOW is this super brittle. Makes me nervous